os_windows_test.go

```diff
@@ -0,0 +1,82 @@
+// +build windows
+
+package ctxvfs
+
+import (
+	"testing"
+)
+
+func TestWindowsMount(t *testing.T) {
+
+	// keys are mount points,
+	// values contain expected results for osFS.resolve(key)
+	tests := map[string]map[string]string{
+		"": {
+			"C:/folder/file":  "C:\\folder\\file",
+			"D:/folder/file":  "D:\\folder\\file",
+			"/C:/folder/file": "C:\\folder\\file",
+			"/folder/file":    "",
+			"file":            "",
+			"/file":           "",
+		},
+		"C:": {
+			"C:/folder/file":  "C:\\folder\\file",
+			"/C:/folder/file": "C:\\folder\\file",
```

Comment:
why are you allowed to specify `C:/` and `/C:/`. Surely you should only be able to specify paths relative to `C:`? I expect the above to cases to resolve to the empty string.

Does the old `vfs.OS("C:")` have this behaviour? This crops up a few times in the test case, I'll just simply comment on them, but refer to this inline comment.